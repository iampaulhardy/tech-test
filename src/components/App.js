import React, { useState } from "react";
import Search from "./Search";
import SearchResults from "./SearchResults";
import "../styles/App.css";

const App = () => {
  const [searchResults, setSearchResults] = useState([]);

  return (
    <div className="App">
      <img 
        className="nasa-logo"
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
        alt="Nasa logo">
      </img>
      <Search setSearchResults={setSearchResults}/>
      <SearchResults results={searchResults}/>
    </div>
  );
};

export default App;
