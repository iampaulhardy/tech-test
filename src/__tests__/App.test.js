import React from "react";
import { render } from "@testing-library/react";
import App from "../components/App";

describe("Search", () => {
  const { asFragment } = render(<App />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
