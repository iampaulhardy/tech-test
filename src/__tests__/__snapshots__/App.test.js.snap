// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="Nasa logo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <div
      class="search"
    >
      <form
        class="search-form"
      >
        <input
          class="search-input"
          type="text"
        />
        <button
          class="search-button"
          type="submit"
        >
          Go!
        </button>
      </form>
    </div>
    <p>
      No results
    </p>
  </div>
</DocumentFragment>
`;
